src/Temporalio/Bridge/src/runtime.rs

```diff
@@ -204,45 +224,49 @@ impl TryFrom<&TelemetryOptions> for CoreTelemetryOptions {
     }
 }
 
-impl TryFrom<&MetricsOptions> for Arc<dyn CoreMeter> {
-    type Error = anyhow::Error;
-
-    fn try_from(options: &MetricsOptions) -> anyhow::Result<Self> {
-        if let Some(otel_options) = unsafe { options.opentelemetry.as_ref() } {
-            if !options.prometheus.is_null() {
-                return Err(anyhow::anyhow!(
-                    "Cannot have OpenTelemetry and Prometheus metrics"
-                ));
-            }
-            // Build OTel exporter
-            let mut build = OtelCollectorOptionsBuilder::default();
-            build
-                .url(Url::parse(&otel_options.url.to_str())?)
-                .headers(otel_options.headers.to_string_map_on_newlines())
-                .metric_temporality(match otel_options.metric_temporality {
-                    OpenTelemetryMetricTemporality::Cumulative => MetricTemporality::Cumulative,
-                    OpenTelemetryMetricTemporality::Delta => MetricTemporality::Delta,
-                })
-                .global_tags(options.global_tags.to_string_map_on_newlines());
-            if otel_options.metric_periodicity_millis > 0 {
-                build.metric_periodicity(Duration::from_millis(
-                    otel_options.metric_periodicity_millis.into(),
-                ));
-            }
-            Ok(Arc::new(build_otlp_metric_exporter(build.build()?)?))
-        } else if let Some(prom_options) = unsafe { options.prometheus.as_ref() } {
-            // Start prom exporter
-            let mut build = PrometheusExporterOptionsBuilder::default();
-            build
-                .socket_addr(SocketAddr::from_str(prom_options.bind_address.to_str())?)
-                .global_tags(options.global_tags.to_string_map_on_newlines())
-                .counters_total_suffix(prom_options.counters_total_suffix)
-                .unit_suffix(prom_options.unit_suffix);
-            Ok(start_prometheus_metric_exporter(build.build()?)?.meter)
-        } else {
-            Err(anyhow::anyhow!(
-                "Either OpenTelemetry or Prometheus config must be provided"
-            ))
+fn create_meter(options: &MetricsOptions, custom_meter: Option<CustomMetricMeterRef>) -> anyhow::Result<Arc<dyn CoreMeter>> {
+    // OTel, Prom, or custom
+    if let Some(otel_options) = unsafe { options.opentelemetry.as_ref() } {
+        if !options.prometheus.is_null() || custom_meter.is_some() {
```

Comment:
Personally I'd probably convert the C repr to a real enum before passing it in here so this validation logic can live by itself nice and separate from the real init stuff, but, up to you, a matter of taste.

Reply:
Somewhere I have to have the mutual-exclusivity validation, whether in here or at enum creation time before here (and the added complexity of requiring that custom meter is _always_ created so it can be dropped even if it fails validation makes this a bit confusing)